regression/pubsub.js

```diff
@@ -91,45 +104,21 @@ describe('pubsub', function() {
   });
 
   describe('Subscription', function() {
-    var TOPIC_NAME = 'test-topic';
+    var TOPIC_NAME = generateTopicName();
     var subscriptions = [
       {
-        name: 'sub1',
+        name: generateSubName(),
         options: { ackDeadlineSeconds: 30 }
       },
       {
-        name: 'sub2',
+        name: generateSubName(),
         options: { ackDeadlineSeconds: 60 }
       }
     ];
     var topic;
 
-    function deleteAllTopics(callback) {
-      pubsub.getTopics(function(err, topics) {
-        if (err) {
-          callback(err);
-          return;
-        }
-        async.parallel(topics.map(function(topic) {
-          return topic.delete.bind(topic);
-        }), callback);
-      });
-    }
-
-    function deleteAllSubscriptions(callback) {
-      pubsub.getSubscriptions(function(err, subs) {
-        if (err) {
-          callback(err);
-          return;
-        }
-        async.parallel(subs.map(function(sub) {
-          return sub.delete.bind(sub);
-        }), callback);
-      });
-    }
-
     before(function(done) {
-      async.parallel([deleteAllTopics, deleteAllSubscriptions], function(err) {
+      deleteAllTopics(function(err) {
```

Comment:
This is the troubled `before` hook. It used to delete topics and subscriptions in parallel, which doesn't seem too smart in retrospect. Especially since I know now that deleting a topic will delete any subscriptions to it.